src/Compilers/VisualBasic/Test/Emit/Emit/EmitMetadata.vb

```diff
@@ -973,5 +975,130 @@ End Class
             Dim options = TestOptions.ReleaseExe.WithPlatform(platform)
             CompileAndVerify(comp.WithOptions(options))
         End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub EmitMetadataOnly_Exe()
+            CompileAndVerify(
+                <compilation>
+                    <file>
+Module Program
+    Sub Main()
+        System.Console.WriteLine("a")
+    End Sub
+End Module
+                    </file>
+                </compilation>,
+                options:=TestOptions.ReleaseExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                emitOptions:=EmitOptions.Default.WithEmitMetadataOnly(True),
+                symbolValidator:=Sub(m)
+                                     Assert.NotEqual(0, m.GetMetadata().Module.PEReaderOpt.PEHeaders.CorHeader.EntryPointTokenOrRelativeVirtualAddress)
+                                     Dim main = m.GlobalNamespace.GetMember(Of MethodSymbol)("Program.Main")
+                                     Assert.Equal(Accessibility.Public, main.DeclaredAccessibility)
+                                 End Sub
+            ).VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub EmitMetadataOnly_Exe_AsyncMain()
```

Comment:
>EmitMetadataOnly_Exe_AsyncMain [](http://example.com/codeflow?start=19&length=30)

Lets keep this test, but I was looking for a scenario that would produce ```ERRID.ERR_AsyncSubMain``` error

Reply:
Adding the test. Turns out the VB's GetEntrypoint can return non-null result plus some error diagnostics, so updating the code a bit to check for that as well.